### DISCUSSION TITLE
Project dead?

### DISCUSSION THREAD

**Discussion by aadithya96** [solokeys/solo2#174] [Q&A]
After waiting for more than a year to get this delivered, it seems like development has slowed died off too much like the hype around this. It seems like the last commit was made more than a year ago. Has anyone heard anything about this or is this project dead?

**Comment by msiqx9** [comment]
Yes, the project is dead.  My v1 and v2 keys still work ok.

**Reply by alexjmoore** [reply]
Unfortunately it does seem dead. For me, my solo2 keys simply don't work on Android via USB or NFC, it's a real shame as I was a big fan of the project originally. For something so key to a digital ecosystem you need to be able to trust it will work when you need it and it seems that simply isn't the case for solo2.

**Reply by digital-idiot** [reply]
I tried to contact them at [hello@solokeys.com](mailto:hello@solokeys.com) regarding the NFC issue. I never heard back from over a month. I contacted them requesting replacement as it seems like manufacturing defect but still no response. After the whole delay fiasco these kinds of things are simply unacceptable.

**Reply by BarsMonster** [reply]
@alexjmoore @digital-idiot There were comments in the indiegogo that to make NFC work one would need to disable pin and it's android limitation, not solo's. Unfortunately I don't know how to disable pin without resetting the key.

**Reply by digital-idiot** [reply]
@BarsMonster  Can you link to the Indiegogo comment? Also if that is the case why mention it somewhere officially, like FAQ or something?

**Reply by digital-idiot** [reply]
The strange thing is for me the NFC do work with Android if the Key is powered separately through USB. Also several users reported that NFC works with Android for their keys as expected.

Example: https://github.com/solokeys/solo2/discussions/165#discussioncomment-7014099

**Reply by BarsMonster** [reply]
@digital-idiot Comment is on second page of user comments at the moment, not possible to link spcific user comment:

[Aldo Bleeker ](https://www.indiegogo.com/individuals/30392049)1 month ago

Backer #2845, my USB A arrived today. It was scheduled to arrive while I was on holiday, but Solokeys held it until I had returned. That’s great service!
I’ve tested it with webauth.me, and webauthn.io, and it has worked without a hitch. I’ve also added it to my Google account, and that worked too. I can even use it on my phone over USB, and NFC. (Remove the PIN first when you try to register it on Google, or you won’t be able to enter a name. I just added the PIN afterwards again.)

[Aldo Bleeker](https://www.indiegogo.com/individuals/30392049) 18 days ago
I've removed the PIN with Windows Settings > Accounts > Sign-in options > Security key > Manage (touch key) > Reset Security Key. I'm assuming this will not just remove the PIN, but reset the entire key, and remove all data. That's fine by me, but this may not apply to others.

**Comment by BarsMonster** [comment]
@aadithya96 I got mine delivered like a month ago, so they are going through backlog. One of the latest updates was that team had to get full time jobs elsewhere, but they still trying to deliver everything. Also, it might be possible to backport features from NitroKey firmware, which is somewhat compatible as far as I understand.

**Comment by BarsMonster** [comment]
The question is is there anyone  in the community who is interested to fork and/or take leadership in firmware development? :-)

**Reply by purepani** [reply]
I think this project is exceptionally cool. I know absolutely nothing about firmware development, and very little about hardware development, but I might just spend the time hacking with this stuff anyway.

**Comment by jugendhacker** [comment]
So I finally got my 2C too, sadly it seems to be completely dead now :(

Would have been better if I had selected NitroKey from the beginning on

**Reply by msiqx9** [reply]
I suggest you contact Solo to get a warranty replacement. They seem to have had reliability problems in manufacturing the USB-C key.

**Reply by jugendhacker** [reply]
@msiqx9 oh I didn't mean dead in the sense of hardware (although I'll probably try a warranty replacement because the connector is dent and it makes strange crackling sound on disconnect) but rather software wise

**Reply by savely-krasovsky** [reply]
They sent me replacement of Solo 2C, though I didn't receive it yet.

**Reply by savely-krasovsky** [reply]
@jugendhacker mine also does crackling sound on disconnect, but works perfectly. I don't understand what it cracking though...

**Reply by rugk** [reply]
@savely-krasovsky did you receive it yet? :eyes:

**Reply by savely-krasovsky** [reply]
@rugk yes, I received the replacement key and it works.

**Comment by sbusch** [comment]
Strange crackling is a good description :-)

I have this sound for both of my USB-C Solokeys. But despite this (and the known software bugs) they still work.

**Reply by jugendhacker** [reply]
Yeah, it works, but still kind of bugged, see [solo2-cli#108](https://github.com/solokeys/solo2-cli/issues/108) for example. And also there were more features promised for the successful Indiegogo campaign (like the PGP smart-card for example), which seems like they will never get developed (at least not by solokeys themselves)

**Comment by schildbach** [comment]
I'm a bit more concerned about the USB-C-Ports I plug my Solo2 into.

**Reply by jugendhacker** [reply]
Yeah, sounds like a condensator going rogue or something

**Reply by sbusch** [reply]
Never thought this could come from the **port** , I always suspected the key 👀 

And yes, since there are no noticable loose parts I really could be due to the electronic.

**Reply by schildbach** [reply]
I'm pretty sure it comes from the key, but I'm afraid it could damage the port.

**Reply by BarsMonster** [reply]
Crackling noise must be from ceramic capacitor. They do emit sound on sharp voltage gradients. This alone is not dangerous and not a defect. If key does not get hot during work (for example due to short circuit) - I would not worry about it.

**Comment by sbusch** [comment]
@jugendhacker I thought about Nitrokey this too, but was reluctant because they used the same software framework as Solo:

https://github.com/Nitrokey/nitrokey-3-firmware

> The Nitrokey 3 firmware is written in Rust. It uses the Trussed firmware framework and is developed in collaboration with SoloKeys (see the solo2 repository).

But Nitrokey has recent updates.

Are your projects still related @robin-nitrokey @sosthene-nitrokey

**Reply by robin-nitrokey** [reply]
Solo2 and Nitrokey 3 share many firmware components that we collaborate on, but the hardware and firmware are developed independently. For questions regarding the Nitrokey 3, please use the [nitrokey-3-firmware repository](https://github.com/Nitrokey/nitrokey-3-firmware).

**Reply by sbusch** [reply]
Thanks for your response!

Is there still activity in your collaboration with Solo? I you don't want to talk about this, no problem...

**Reply by BarsMonster** [reply]
I wonder if it is possible to backport Nitrokey 3 software to Solokey...

**Reply by savely-krasovsky** [reply]
@BarsMonster the problem is that even if it's possible, only Hacker Edition could be reflashed. Otherwise Solo team will need to sign it by their keys to allow us to update.

**Comment by nyabinary** [comment]
rip ig :<

**Reply by msiqx9** [reply]
Now that I'm into using passkeys, I haven't needed to use my physical Solo/Yubico keys for months.

**Comment by kenjikato** [comment]
I was just about to switch over to using the Solo v2 keys that I purchased (4 of them back during the kickstarter) only to come here and find out this project is effectively dead. 🤬
I love the idea of a hardware key, but I have now backed 5 different projects over the last 10ish years with all but one of them having died.
At this point I would be happy to pay a yearly "maintenance" fee if it would keep at leas one of these projects going beyond the initial production stage of the product.
Ugh! 🙄

**Comment by theerror** [comment]
Yep, it's indeed sad. I wondering, if Nitrokey actually working better, or it have same problems with support and development. Anyway, seems Solo is dead indeed.

**Comment by kenjikato** [comment]
My current goto hardware key and the only one that is still fully supported out of all the hardware keys I've backed is the [OnlyKey](https://onlykey.io).
My goal with the Solo v2 was to use it for FIDO access for work and give several of my co-workers their own keys to access the same services.
This way I could use my OnlyKey for my personal passwords and things like SSH keys access. Keeping my work and personal stuff separated nicely.
At this point I'm just going to go buy multiple more OnlyKeys (I like the Duo but the pin touch zones are a bit hard to hit sometimes without an extension cable from my computer) and keep using them. I'm not even going to mess with the Nitrokey at this point.

**Reply by nyabinary** [reply]
May I recommend [tillitis key?](https://tillitis.se/), it's a spinoff company from Mullvad VPN so it's pretty reputable.

**Reply by kenjikato** [reply]
Thanks for the recommendation. It looks cool but it doesn't do FIDO/U2F yet (unless I missed something.) Which was going to be my main usage purpose for the Solo v2. I can get FIDO 2 / U2F from OnlyKey already.

**Comment by msiqx9** [comment]
In recent months I've not used physical keys; I'm now relying solely on passkeys stored in Bitwarden Password Manager and in Apple iCloud.  Passkeys work really well, they are based on FIDO2 and they are free.

**Reply by BarsMonster** [reply]
You just need to trust Apple on that ) Not your physical key - not your... passwords :-)

**Reply by synthgab** [reply]
It's funny, I do exactly the opposite: I never really used my Solo 2 except from time to time, until a few months ago when passkeys became a trend and all websites ask to register one.

Ok, maybe I'm over paranoid, but I don't feel confident with having logins, passwords and passkeys stored in the same database/device...

[Edit] : typo

**Comment by msiqx9** [comment]
There seems to be some code [activity](https://github.com/solokeys/solo2/commit/bf0e51fc44daedb7b8f9edd6fb9e27b00d2fd319) yesterday.

**Reply by savely-krasovsky** [reply]
Lol, seems we finally will get an update.

**Reply by savely-krasovsky** [reply]
Their [X account](https://x.com/SoloKeysSec) seems to be active too. [Patrick Connor](https://x.com/_conorpp) reposts them too. So it seems like there is definitely some activity.

**Reply by savely-krasovsky** [reply]
There is again some activity: https://x.com/SoloKeysSec/status/1830348162462261459
They also kind confirmed that development didn't stop.

**Reply by rugk** [reply]
Maybe they also should get off Twitter… time has passend and things™ happened…

**Reply by manuel-domke** [reply]
Imho, this kind of negativity might be better suited for Twitter, where it's easy to vent frustrations.

I understand that some people had, and still have, higher expectations. Open-source projects and crowdfunding ventures often have their ups and downs, and SoloKeys is no exception. The project is clearly going through a tough phase, but labeling it as dead doesn't help anyone.

As someone who still believes in this project and is actively working to move it forward, I find comments like this disheartening. 
Instead of adding to the negativity, why not think about how we can support the project...

The team and community are doing their best with the resources they have. Let's channel our energy into something constructive rather than tearing each other down.

**Reply by purepani** [reply]
I would like to point out that labeling as dead does in fact help people make informed purchasing and contribution investment decisions if the project is in fact dead(which it certainly looked like it was), but otherwise I fully agree.

**Reply by manuel-domke** [reply]
True. But from what I know, work is still being done on the USB-C version to address the issues discussed earlier in this thread. But hardware production of such a product isn't an easy task.

While the project has faced challenges, the basic functionality of an open-source FIDO2 stick works as intended. FIDO2 attestation remains an open task, but it's something the team is actively working on.

So while it's not perfect, I wouldn't call it dead just yet. There's still progress being made, even if it's slower than some might have hoped.

**Reply by savely-krasovsky** [reply]
@manuel-domke while I agree on overall point of your messages, I wouldn't agree that "the team are doing their best". Surely I hope they finally have some time to resume their work on firmware and probably complete FIDO certification, but their complete lack of communication for almost entire year is not what I would expect from the project.

I totally understand that they have faced with some problems, need to work and earn money other ways, but instead of just ghosting the community they could write a blog post explaining the situation, difficulties they faced and realistically what is the future of the project. I would accept anything they would wrote, because the risks was known, but instead they were radio silent all that time until recently.

Of course I am still happy to see some activity now, but it's still would be nice to see the explanation with at least their hypothetical roadmap.

**Comment by hoos97** [comment]
Hey guys...figured I'd drop in to give some color to what has gone on as the most active member of the SoloKeys team.  Unfortunately, I am not a developer!  I've handled all of the behind the scenes works and communications.
Everyone remembers we launched Solo V2 (now labeled as Solo 2) the year following covid.  The impacts to supply chain were significant and made delivery very challenging for crowdsourcing campaigns.  However, we truly struggled with USB-C, and our manufacturer, and our PCB supplier.  We also needed employment as SoloKeys, intended to be our number 1 thing, wasn't ready to be our number 1.  As a result, our team was onboarded by another company to build an entirely separate product.  The payer of bills got the lion share of our work.  Developer hours are our most constrained resource.

So, business operations have simmered, development has simmered.  And capital mostly used up to source machinery to resolve USB-C issues.

We have a solution in place for USB-C.  Manufacturing is figured out.  And crowdsourcing is completed.  Stores are reopening in Europe and US.  Product is ready for the sales cycle and can satisfy the majority of the 2FA market.  So, the firmware enables product sale for the average user.  

Meanwhile, we are partnered with NitroKey to further firmware development.   This progress can be seen at https://github.com/trussed-dev.  As cash  flow goes positive and we benefit from sales cycle, dev work will start to increase again at solokeys, to include porting updates from trussed in order to complete tests for the FIDO certification.  After that, we will expect to use cash flow to build in developer work for some more of the features that is expected of the product.  
I greatly appreciate everyone's patience...it has been a long 3 years. 

I will say that SoloKeys is a completely bootstrapped team.  All moneys have come from crowd sourcing campaigns, some development funds and even our own pockets for the past year.  We have a product that is ready...if you know anyone that wants to contribute to the code base...get the word out.  Even better, if you know someone who wants to invest in a product that is now ready to sell...get the word out!

**Reply by savely-krasovsky** [reply]
I am very pleased to hear this. Do I understand correctly that your overall plan is to port all Nitrokey team improvements on Trussed framework to Solo 2 family, but for now you didn't really start any port-related development and realistically we cannot wait the real update any time soon?

**Reply by piratenpanda** [reply]
Thanks for the informations! That's good to know but this should have been posted years ago, keeping the backers informed and not left totally clueless. It's nice to know though that the project still lives!

**Reply by hoos97** [reply]
Duly chastised.  Regarding the Trussed framework: Trussed is the shared subset of the Solo and Nitro firmware. We developed it and Nitro has since extended it.  We have benefitted from an open relationship with the Nitro development team for the last few years.  When we are ready, we will update our firmware pulling from Trussed. We have testing to do our on our end before that happens.

**Reply by savely-krasovsky** [reply]
I would happy to test pre-release versions with my hacker edition key!

**Reply by purepani** [reply]
Heavily appreciate the communication after the response in the previous thread!
I really think the most important thing is somewhat periodic updates on what's going, even if it's a vacuous update when things are slow, to at the very least know that someone on the team still cares about the project progressing forward. 
As an example, despite liking solokey, I had specifically chosen not to contribute and not to recommend the key to someone else because there had been zero activity as far as I was able to tell, and it felt like any work would have gone nowhere, despite really wanting to believe in the project.

I have had a lot on my plate recently, so I might not be able to, but if there is anywhere to contribute to development at the moment, I'd be glad to at least take a look and add that to my (ever increasing) task list!

**Reply by synthgab** [reply]
Honestly, this is the best update since the crowd sourcing campaigns, the one we all expected during the past couple of years, which is an amazing one considering the news it brings regarding the future of Solo 2.

On a side note, considering all you went through, I am glad I chose Limited Editions Solo 2; it already casts some kind of 'special device' sparks, but in light of this update, it will step ahead to the status of a unique artifact I'll be proud to own.

Thank you again for all your efforts, please do not give up, despite the difficulties, at least for the USB-A form factor, you really managed to craft a robust and built-to-last product !!

**Reply by j4lc** [reply]
While it's good to see some activity again, there's still one question I must ask: why wasn't there any blog or social media post saying "It's slowing down, we're sorry, but we're gonna come back later if we can"?

This is honestly what made me lose faith in the whole project, the lack of communication. I know there's only so much time, but even a quick post could have made things much better.

**Reply by rugk** [reply]
I agree, especially as during the kickstarter campaign, we had such updates frequently, at some time then it stopped…

**Reply by andrebrei** [reply]
I'm super happy to see that update too. In fact I had given up on the keys (one of my requirements was that it works of Microsoft entra/aad) because there was so little comms/info for such a long time.

I returned here because I spotted the solo and somu are on Microsoft's attestation list now. So clearly there was some progress.
(obviously, it won't work with the hacker version but I'll have go with a normal solo/somu once I'm back from travel).
edit: yes, I noticed, the list entry is for the solo v1 not 2.

**Reply by savely-krasovsky** [reply]
AFAIK Solo v1 has FIDO2 attestation since it release, so I don't think it's a recent change. The main problem is Solo v2...

**Reply by shrimpwtf** [reply]
Does this mean we'll need to move to trussed's firmware at a later point?

**Reply by savely-krasovsky** [reply]
Current Solo2 firmware is already built on top of Trussed. It just uses an older version of framework without a lot of newer features and bug fixes which Nitrokey team added and fixed later.

**Reply by a55uka** [reply]
so at this immediate moment is it okay to assume getting a nitrokey would be okay? im slightly off put by its reputation and actions in the past

**Reply by mkopec** [reply]
> if you know anyone that wants to contribute to the code base...get the word out.

@hoos97  Is there a list of issues for newcomers who would like to help? :)

**Reply by savely-krasovsky** [reply]
It's almost half a year, is there any news on porting and certifying? Yubico and Token2 recently got FIDO Level 2 certification.

**Reply by AlexeyMatskevich** [reply]
any news? lately keys stopped working anywhere except linux, on android and mac just endless errors when trying to authorize in password managers, but on some sites it works.

**Reply by msiqx9** [reply]
This project died two years ago.  I’m surprised the keys are still being offered for sale.  Nowadays I largely use Bitwarden password manager for passkeys and [Yubico](https://www.ebay.co.uk/itm/405886266408) for backup.

**Reply by schildbach** [reply]
The Solo2 keys work fine, I use them both as a second factor and as a passkey dozends of times every day. So why not keep selling them?

**Reply by icedream** [reply]
Any further updates? Personally, I'm waiting on a firmware update that would implement full PIV or PGP support and have been eagerly looking towards the NitroKey firmware for that, especially given the constant mentions of Trussed as the common factor - but the community forum posts over there for getting their firmware running on Solo are rather sparse at best, skipping over any useful details.

**Reply by piratenpanda** [reply]
Write it off. This has been an utter disappointment

**Reply by savely-krasovsky** [reply]
I personally went with Token2. Their recent firmware has both OpenPGP and PIV + FIDO L2 certification. And they cost a fraction of Yubikeys.

**Comment by raouf-haddada** [comment]
DO NOT update,
And my key is completely broken
$ solo2 update
Downloading latest release from https://github.com/solokeys/solo2/
Fetched firmware version 2:20220822.0 (2.964.0)
LPC55 Bootloader detected. The LED should be off.
Writing new firmware...
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████░░ 308704/310672Done. Rebooting key. The LED should turn back on.

Nothing happens, no reboot, I can only get access to bootloader, no way technically to build a new firmware.
a rollback with old firmware is impossible ::
$ solo2 update -w solo2-firmware-1.0.9.sb2
Fetched firmware version 1:20200101.9 (1.0.9)
LPC55 Bootloader detected. The LED should be off.
Writing new firmware...
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████░░ 314784/316816aborting
thread 'main' panicked at 'unexpected status SbLoader(RollbackBlocked)', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/lpc55-0.2.1/src/bootloader/protocol.rs:299:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

My solo2 key device is completely broken, not usable

Sadly to confirm that the project is completely dead. no mail support answers.
So if the product is bought, no support guaranteed, at all.

**Comment by jugendhacker** [comment]
Looks like there is some development activity again? A lot of commits last week...

**Comment by hoos97** [comment]
Yeah...got a little simmer going, with a chance to heat up and get to a boil.  Will keep you posted as we release more news.

**Comment by tradenet** [comment]
Seems there is some Google ads going on. That's what brought me here.

**Reply by hoos97** [reply]
Ads are running.  If you all are reviewing activity, we do have a GUI in the works, and some steps are being taken towards FIDO certification.

**Reply by PeterFalken** [reply]
Sounds good, nice to see the project moving along.

**Reply by mkopec** [reply]
I am very happy to hear that. I haven't been able to use my Solo 2 at my bank because of the missing certification. Would be very nice to see this addressed.

**Reply by nyabinary** [reply]
So when's the new firmware going be out? :P

**Reply by hoos97** [reply]
Just finished our FIDO interoperability testing last week!  Moving through some of the admin processes now and will be able to announce the certification in the near future.  Firmware will be released shortly after.

**Comment by no-usernames-left** [comment]
As someone with one of the original Kickstarter keys, will they also be supported?

**Reply by hoos97** [reply]
Yes, it will be backwards compatible.  The other side of this is a GUI that has been developed.  It can be tested here: https://github.com/leetronics/solo2-gui.  
Bear in mind, this is only valid for Solo 2 Hacker at the moment.  There is a prerelease of software you can obtain to test out. 
If you wanted to run the latest experimental firmware, you would need a hacker device.  When we release the firmware, we hope to release the GUI platform as well to enable people a simple process to update.  But, there will be a means to update via CLI as well.

**Reply by PeterFalken** [reply]
@hoos97, the tools looks good - works on my KS Solo2 keys and also some later ones with the updated USB-C connector. Great to see the project coming together. By the way, what is the maximum capacity (in terms of slots) for resident keys on the Solo2 devices?

**Reply by Forage** [reply]
> When we release the firmware, we hope to release the GUI platform as well to enable people a simple process to update.

If a simple process for updates is the goal, [fwupd](https://github.com/solokeys/solo1/issues/221) would be the way to go for us Linux users ;)

**Reply by piratenpanda** [reply]
@hoos97 +1 for fwupd support

**Reply by savely-krasovsky** [reply]
@PeterFalken for the moment it's 50, but Large Blob storage is disabled right now.

It's worth mention that Solo2 has 1MB unused memory (its firmware doesn't even ship drivers for it rn), but it would be very hard to certify this, IMO, because this external memory would require the use of encryption.

**Reply by PeterFalken** [reply]
Thanks for the info @savely-krasovsky, maybe in the future the unused memory can be used as an expanded storage area. I can see how that would add extra steps into the initialization of the device after a factory reset.

**Comment by 0x0ece** [comment]
Update in this new thread: https://github.com/solokeys/solo2/discussions/205
